src/modules/sensors/vehicle_air_data/VehicleAirData.cpp

```diff
@@ -297,27 +297,66 @@ void VehicleAirData::Run()
 
 float VehicleAirData::PressureToAltitude(float pressure_pa, float temperature) const
 {
-	// calculate altitude using the hypsometric equation
-	static constexpr float T1 = 15.f - CONSTANTS_ABSOLUTE_NULL_CELSIUS; // temperature at base height in Kelvin
-	static constexpr float a = -6.5f / 1000.f; // temperature gradient in degrees per metre
-
-	// current pressure at MSL in kPa (QNH in hPa)
-	const float p1 = _param_sens_baro_qnh.get() * 0.1f;
-
-	// measured pressure in kPa
-	const float p = pressure_pa * 0.001f;
-
-	/*
-	 * Solve:
-	 *
-	 *     /        -(aR / g)     \
-	 *    | (p / p1)          . T1 | - T1
-	 *     \                      /
-	 * h = -------------------------------  + h1
-	 *                   a
-	 */
-	float altitude = (((powf((p / p1), (-(a * CONSTANTS_AIR_GAS_CONST) / CONSTANTS_ONE_G))) * T1) - T1) / a;
-
+    float altitude = 0.0f;
+
+    if (_param_baro_medium.get() == 0)
```

Comment:
you could add an enum in the header file like that to make it more readable
```
enum class Medium {
   Air,
   FreshWater,
   SaltWater
}
```
```suggestion
    if (_param_baro_medium.get() == Medium::Air)
```

Reply:
3670f4722d16f331a9b23a5c6910b4a5c062fdea